### PR TITLE
[#172038263] Add Express Error Middleware, catch passport exceptions

### PR DESCRIPTION
**Pull request by BurnedMarshal**

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -199,9 +201,12 @@ export function newApp(
...
 
   // Initiliaze Url Token Authenticator
-  const urlTokenAuth = passport.authenticate("authtoken", {
-    session: false
-  });
+  const urlTokenAuth = middlewareCatchAsInternalError(
+    passport.authenticate("authtoken", {
+      session: false
+    }),
+    "An expection occurs on Auth Token Strategy"
```
Comment (gunzip) [inline src/app.ts:208]:
```suggestion
    "An exception occurred authenticating with Auth Token"
```

File: src/app.ts [hunk 2]
```diff
@@ -321,9 +328,12 @@ function registerPagoPARoutes(
...
   profileService: ProfileService
 ): void {
-  const bearerWalletTokenAuth = passport.authenticate("bearer.wallet", {
-    session: false
-  });
+  const bearerWalletTokenAuth = middlewareCatchAsInternalError(
+    passport.authenticate("bearer.wallet", {
+      session: false
+    }),
+    "An Exception occurs on Bearer Wallet Token Strategy"
```
Comment (gunzip) [inline src/app.ts:335]:
```suggestion
    "An exception occurred authenticating with Bearer Wallet Token"
```

File: src/app.ts [hunk 3]
```diff
@@ -353,9 +362,12 @@ function registerAPIRoutes(
...
   userDataProcessingService: UserDataProcessingService
 ): void {
-  const bearerSessionTokenAuth = passport.authenticate("bearer.session", {
-    session: false
-  });
+  const bearerSessionTokenAuth = middlewareCatchAsInternalError(
+    passport.authenticate("bearer.session", {
+      session: false
+    }),
+    "An Exception occurs on Bearer Session Strategy"
```
Comment (gunzip) [inline src/app.ts:369]:
```suggestion
    "An exception occurred authenticating with Bearer Session Token"
```

File: src/app.ts [hunk 4]
```diff
@@ -536,15 +548,17 @@ function registerAPIRoutes(
...
   acsController: AuthenticationController
 ): void {
-  const bearerTokenAuth = passport.authenticate("bearer.session", {
-    session: false
-  });
+  const bearerTokenAuth = middlewareCatchAsInternalError(
+    passport.authenticate("bearer.session", {
+      session: false
+    }),
+    "An Exception occurs on Bearer Token Strategy"
```
Comment (gunzip) [inline src/app.ts:560]:
```suggestion
    "An exception occurred authenticating with Bearer Session Token"
```

File: src/strategies/bearerSessionTokenStrategy.ts [hunk 5]
```diff
@@ -29,10 +29,20 @@ const bearerSessionTokenStrategy = (
     ) => {
```
Comment (gunzip) [inline src/strategies/bearerSessionTokenStrategy.ts:29]: what about bearerWalletTokenStrategy ?

File: src/utils/middleware/express.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+ * Catch an express error and returns a ResponseErrorInternal response
+ * @ref https://expressjs.com/en/guide/error-handling.htmls
+ */
+export function expressErrorMiddleware(
+  err: Error,
+  _: Request,
+  res: Response,
+  __: NextFunction
+): void {
+  log.error("Errors middleware catchs an error: %s", err.message);
```
Comment (gunzip) [inline src/utils/middleware/express.ts:16]:
```suggestion
  log.error("An exception occurred during http request: %s", err.message);
```

File: src/strategies/bearerSessionTokenStrategy.ts (hunk 5 above)
Comment (BurnedMarshal) [reply src/strategies/bearerSessionTokenStrategy.ts]: Fixed in 67317c1

File: src/utils/strategies.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+ * with proper parameters depending on the response of
+ * methods getBySessionToken or getByWalletToken.
+ */
+export function fulfill(
+  errorOrUser: Either<Error, Option<User>>,
+  done: StrategyDoneFunction
+): void {
+  errorOrUser.fold(
+    error => done(undefined, false, error.message),
+    user => done(undefined, user.isNone() ? false : user.value)
```
Comment (gunzip) [inline src/utils/strategies.ts:25]:
```suggestion
    user => done(undefined, user.getOrElse(false))
```

File: src/utils/strategies.ts [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+ * with proper parameters depending on the response of
+ * methods getBySessionToken or getByWalletToken.
+ */
+export function fulfill(
+  errorOrUser: Either<Error, Option<User>>,
+  done: StrategyDoneFunction
+): void {
+  errorOrUser.fold(
+    error => done(undefined, false, error.message),
+    user => done(undefined, user.getOrElse(false))
```
Comment (gunzip) [inline src/utils/strategies.ts:25]:
```suggestion
    user => done(undefined, user.isNone() ? false : user.value)
```

**Pull request merged**